test/system/home_test.rb

```diff
@@ -14,32 +14,101 @@ class HomeTest < ApplicationSystemTestCase
     assert_equal 'ダッシュボード | FJORD BOOT CAMP（フィヨルドブートキャンプ）', title
   end
 
-  test 'GET / without github account ' do
+  test 'verify message presence of github_account registration' do
     visit_with_auth '/', 'hajime'
-    within('.card-list__item-link.is-github_account') do
-      assert_text 'GitHubアカウントを登録してください。'
-    end
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_text 'GitHubアカウントを登録してください。'
+
+    # GitHub との連携処理を update! で代用
+    users(:hajime).update!(github_account: 'hajime')
+    visit '/'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_no_text 'GitHubアカウントを登録してください。'
   end
 
-  test 'GET / with github account' do
-    user = users(:hajime)
-    user.update!(github_account: 'hajime')
+  test 'verify message presence of discord_account registration' do
     visit_with_auth '/', 'hajime'
-    assert_no_selector '.card-list__item-link.is-github_account'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_text 'Discordアカウントを登録してください。'
+
+    visit '/current_user/edit'
+    fill_in 'user[discord_account]', with: 'hajime#1111'
+    click_button '更新する'
+    visit '/'
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_no_text 'Discordアカウントを登録してください。'
   end
 
-  test 'GET / without discord_account' do
+  test 'verify message presence of avatar registration' do
     visit_with_auth '/', 'hajime'
-    within('.card-list__item-link.is-discord_account') do
-      assert_text 'Discordアカウントを登録してください。'
-    end
+    assert_selector 'h2.page-header__title', text: 'ダッシュボード'
+    assert_text 'ユーザーアイコンを登録してください。'
```

Comment:
好みの問題かもですが、このようにすると▲1行になって、確度や可読性が上がるような気がしました〜
```suggestion
    assert_selector '.card-list__item-link.is-avatar_attached', text: 'ユーザーアイコンを登録してください。'
```
あと、こちらをテストで踏襲されない意図がありましたら、ご教示ください〜🙏
> ユーザー情報登録フォームでの各項目の表示順と合うように並び替え

Reply:
@aim2bpg
コメントありがとうございます！
まず、このように書いた経緯を説明させてください🙏

### commit : [テストを修正](https://github.com/fjordllc/bootcamp/pull/4300/commits/f6721b0d6ff9326b5438c0b73d306f2ae76057aa) 

こちらのコミットで「メッセージがないこと」を確認する際に該当のページにいることを証明するために、 `assert_selector 'h2.page-header__title', text: 'ダッシュボード'` を `assert_no_text` の前に追加しました。
その対比としてわかりやすいように（該当のページにいることを証明する目的も含む）、「メッセージがあること」を確認する際にも同じアサーションをしていました。

この方法は、以下のPRの `test/system/questions_test.rb:110` 以降の修正を参考にしました。
[wait\_for\_vuejsを可能な限り排除する by JunichiIto · Pull Request \#4333 · fjordllc/bootcamp](https://github.com/fjordllc/bootcamp/pull/4333/files#diff-893e3ac874f2fc0e2dba9deccac531e7cd550ed9ae798488321d7541a5c7da1b)

また、目的のメッセージの確認に、`assert_text` を使っている理由は以下です。（これも好みがあるとは思いますが...）

- ページ内の他のテキストと被っていない（他のデータの変更などがあっても、被る可能性はかなり低いと思う）
- ぱっと見た時に、テストしている内容がわかりやすい（と思う）

`assert_selector '.card-list__item-link.is-avatar_attached', text: 'ユーザーアイコンを登録してください。'`
この一行でいいのでは？ ということも理解できるのですが、経緯を踏まえた上での意見も聞かせていただけるとありがたいです🙇‍♂️ （初めから説明できておらず、申し訳ありません。）

> あと、こちらをテストで踏襲されない意図がありましたら、ご教示ください〜🙏

こちらに関しては、エラーメッセージの表示方法（モデルのエラーメッセージを順に表示）の関係でモデルの attribute を並び替える必要があったのですが、テストは実際の表示に関係ないので、もとのテストコードと同じGitHubアカウントから書き出していました😅
気になるようでしたら修正したいと思います！（私は、まあいいだろうと思ってしまいました...）

Reply:
ご説明ありがとうございます🙇🏻‍♂️
経緯を理解しましたので、このままでいいと思います👍
自分の提案は厳密すぎてこだわりの範疇だと思いました😅

Reply:
ご説明ありがとうございます🙇🏻‍♂️
経緯を理解しましたので、このままでいいと思います👍
自分の提案は厳密すぎてこだわりの範疇だと思いました😅